### ISSUE TITLE
Use the recipe book to craft multiple stacks of items, resulting in loss of items

### ISSUE THREAD

**Issue by patatoplugin** [PluginBugs/Issues-ItemsAdder#2084]
### Discord tag (optional)

neverlag#6619

### What happened?

If you use recipe book to craft many stacks of items, will lost some

### Steps to reproduce the issue

1.copy my files and create this item
2.just like my video do 

### Server version

Current: git-Purpur-1825 (MC: 1.19.2)*
Previous: git-Purpur-1809 (MC: 1.19.2)
* You are 13 version(s) behind
Download the new version at: https://purpurmc.org/downloads

### ItemsAdder Version

Itemsadder version 3.2.5-beta5（seems not the newest）

### ProtocolLib Version

protocollib 5.0.0-SNAPSHOT-b600

### LoneLibs Version

Lonelibs version 1.0.21

### Other files, you can drag and drop them here to upload. (optional)

```yml
info:
  namespace: dripdrop
items:
  bamboo_tea:
    display_name: 紫颂清竹茶
    lore:
    - '&f♠♠'
    permission: bamboo_tea
    resource:
      material: BREAD
      generate: true
      textures:
      - food/bamboo_tea.png
    events:
      eat:
        feed:
          amount: 4
        give_item:
          item: GLASS_BOTTLE
          amount: 1  
        potion_effect_first:
          type: SPEED
          amplifier: 0
          duration: 600
        potion_effect_second:
          type: SLOW_FALLING
          amplifier: 0
          duration: 600
recipes:
  crafting_table:
    bamboo_tea:
      permission: dripdrop.bamboo_tea
      enabled: true
      pattern:
      - BXX
      - CXX
      - TXX
      pattern_2:
      - XBX
      - XCX
      - XTX
      pattern_3:
      - XXB
      - XXC
      - XXT
      ingredients:
        B: BAMBOO
        T: GLASS_BOTTLE
        C: CHORUS_FRUIT
      result:
        item: dripdrop:bamboo_tea
        amount: 1
```

### Screenshots/Videos (you can drag and drop files or paste links)

https://cdn.discordapp.com/attachments/1040121487132205178/1040121575392956467/1668053598171.mp4

**Comment by SirCraftypants** [comment]
previously known bug that must have broke again, sure lonedev will fix it.

**Comment by LoneDev6** [comment]
Fixed in 3.3.0


https://user-images.githubusercontent.com/27242001/202754665-c89d97ac-bfb0-4099-95a5-85465fd06f07.mp4
